### ISSUE TITLE
Scala 3: completions not shown after dealiasing

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals#3786]
### Describe the bug

Completions don't seem to work on higher-kinded types applied in a type alias. Not sure if this is a Metals issue or something in the compiler, but here it is:

```scala
//> using scala "3.1.1"
//> using option "-Ykind-projector"
package example

trait Service[F[_]] {
  def listStreams(): F[Unit]
}

type Client[Alg[_[_]]] = Alg[Option]

val client: Client[Service] = ???
val client2: Service[Option] = ???

def test = client. /* doesn't work */ listStreams()
def test2 = client2. /* works */ listStreams()
```

Completions on `client.` don't show anything that actually comes from `Service`'s interface, but completions on `client2` work flawlessly.

### Expected behavior

Completions work in both cases

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.2+125-ed5fb2f2-SNAPSHOT

**Comment by tgodzik** [comment]
Thanks for reporting! This looks like something that might need to get fixed in the compiler, I will double check next week.

**Comment by tgodzik** [comment]
Finally managed to double check and it's an issue in the compiler, reported here: https://github.com/lampepfl/dotty/issues/14984

**Comment by tgodzik** [comment]
Looks like this will be fixed in the next compiler version. Let's take a look at it then.

**Comment by tanishiking** [comment]
Closing, confirmed this is fixed in Scala 3.2.0-RC1 https://github.com/lampepfl/dotty/pull/15047 :tada:

**Comment by kubukoz** [comment]
working as expected in 3.2.0!

<img width="847" alt="image" src="https://user-images.githubusercontent.com/894884/194402434-186dc00b-4c21-41e1-95fd-0d551607dc18.png">
